### PR TITLE
Standardize and optimize icon SVG

### PR DESCRIPTION
**Pull request by AndreVallestero**
Optimize icon.svg and base derivative icons on the same SVG path

Resulting optimization:

- `icon.svg` 7,722B -> 797B
- `project_icon.svg` 3,773B -> 849B
- `DefaultProjectIcon.svg` 4,173B -> 921B

Note, DefaultProjectIcon's 128x128 base dimensions are preserved. The other two icons should be resized as needed, so no default dimension was set.

### CODE REVIEW

**Comment by YeldhamDev** [comment]
I really don't like how "icon.svg" looks here. If the intent was to make it follow the aspects of the other icon, I think that one should be modified instead.

**Comment by AndreVallestero** [comment]
@YeldhamDev it's funny that you mention that, since `icon.svg` is the reference so the new version is pretty much identical. Here's a side-by-side:
image

**Comment by YeldhamDev** [comment]
I'm talking about the paddings in the "DefaultProjectIcon.svg" icon.

**Comment by akien-mga** [comment]
Why did the color change? Before `#478cbf`, after `#4488cc`.
The color is part of Godot's branding and shouldn't be changed.

**Comment by AndreVallestero** [comment]
[e4b662c](https://github.com/godotengine/godot/pull/68598/commits/e4b662cb3bf1593cb734138d4d6ff63504ebe2ae) reverts the 12bit color optimization and the padding change noted by @YeldhamDev

**Comment by AndreVallestero** [comment]
@YeldhamDev @akien-mga is there anything blocking this PR from getting approved?

**Comment by aaronfranke** [comment]
@AndreVallestero Are the file size numbers in the OP up-to-date? Can you show before/after images?

**Comment by AndreVallestero** [comment]
icon.svg 7,722B -> 804B
project_icon.svg 3,773B -> 907B
DefaultProjectIcon.svg 4,173B -> 911B

Before and after images can be previewed here: https://github.com/godotengine/godot/pull/68598/files

File: icon.svg [hunk 1]
```diff
@@ -1,139 +1 @@
...
-      <path
-         d="m 0,0 c 0,-7.994 6.477,-14.473 14.471,-14.473 8.002,0 14.479,6.479 14.479,14.473 0,7.994 -6.477,14.479 -14.479,14.479 C 6.477,14.479 0,7.994 0,0"
-         style="fill:#414042;fill-opacity:1;fill-rule:nonzero;stroke:none;stroke-width:0.32031175"
-         id="path108-9"
-         inkscape:connector-curvature="0" />
-    </g>
-  </g>
-</svg>
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 999 999"><g fill="#fff"><path d="M102 657v32q397 345 795 0v-32z"/><path fill="#478cbf" d="M128 232q-46 48-81 106 26 41 55 79v240l149 14q12 2 14 15l4 64 128 10 9-60q2-11 14-14H579q12 3 14 15l8 59 129-9 4-67q4-12 14-13l149-14V417q29-38 55-79-35-57-81-106-42 21-80 46-39-35-86-62 6-50 7-100-56-27-120-41-25 41-45 87-48-7-95 0-19-44-45-87-62 13-120 41 1 50 7 100-46 26-86 62-39-26-80-46M102 689v39c3 272 798 267 795 0v-39l-132 12-5 67q-1 10-13 13l-157 11q-12 1-17-10l-10-64H436l-10 63q-3 10-13 11l-161-11q-10-2-13-12l-5-68Z"/><path d="M471 502v84c3 33 54 33 57 0v-84c-3-32-53-33-57 0"/><circle cx="708" cy="514" r="88"/><circle cx="291" cy="514" r="88"/></g><g fill="#414042"><circle cx="300" cy="520" r="59"/><circle cx="700" cy="520" r="59"/></g></svg>
```
Comment (YuriSizov) [inline icon.svg:1]: I think our SVG library works better if you have fixed width and height in the `<svg>` tag, not just viewBox.

File: misc/dist/project_icon.svg [hunk 2]
```diff
@@ -1 +1 @@
...
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 999 999"><rect fill="#1e1a21" x="8" y="8" width="983" height="983" rx="176" stroke-width="16" stroke="#2e2832"/><g fill="#fff"><path d="M102 657v32q397 345 795 0v-32z"/><path fill="#478cbf" d="M128 232q-46 48-81 106 26 41 55 79v240l149 14q12 2 14 15l4 64 128 10 9-60q2-11 14-14H579q12 3 14 15l8 59 129-9 4-67q4-12 14-13l149-14V417q29-38 55-79-35-57-81-106-42 21-80 46-39-35-86-62 6-50 7-100-56-27-120-41-25 41-45 87-48-7-95 0-19-44-45-87-62 13-120 41 1 50 7 100-46 26-86 62-39-26-80-46M102 689v39c3 272 798 267 795 0v-39l-132 12-5 67q-1 10-13 13l-157 11q-12 1-17-10l-10-64H436l-10 63q-3 10-13 11l-161-11q-10-2-13-12l-5-68Z"/><path d="M471 502v84c3 33 54 33 57 0v-84c-3-32-53-33-57 0"/><circle cx="708" cy="514" r="88"/><circle cx="291" cy="514" r="88"/></g><g fill="#414042"><circle cx="300" cy="520" r="59"/><circle cx="700" cy="520" r="59"/></g></svg>
```
Comment (YuriSizov) [inline misc/dist/project_icon.svg:1]: Same here.

**Comment by YuriSizov** [comment]
I would also make viewBox match the fixed height and width on all three of the images. Other than that, it looks fine.

**Review by MewPurPur — COMMENTED** [review]
Very good, I think this optimized icon is so clean and close to the original that even our official branding stuff (e.g. logo) should use it. However:

- I'm not sure about using a big white overlay for the mouth.
- Yuri's review should be addressed
- It'd be good to have a consistent viewbox and dimensions, or at least a factor of 2/4/8 difference IMO. (edit: Oh wait, Yuri also said this.)

These improvements could also be carried over to:
- Godot.svg
- GodotFile.svg
- GodotMonochrome.svg

**Comment by YuriSizov** [comment]
@MewPurPur You can probably pick up the torch and finish this PR, keeping the original author as a co-author :)

**Comment by MewPurPur** [comment]
Mhm. On a second thought, it's fine to use a BG overlay for the mouth, as long as its edges don't overlap with blue edges.

**Comment by akien-mga** [comment]
Superseded by #78015. Thanks for your contribution!

**Pull request closed without merging**